npm/src/clientfileprocessing/index.ts

```diff
@@ -40,6 +40,30 @@ export class ClientFileProcessing {
             "Percent of metadata extracted: " +
               progressInformation.percentageProcessed
           )
+          const fileUpload: HTMLDivElement | null = document.querySelector(
```

Comment:
I would abstract the progress bar function to a named function instead of having it as an anonymous call back function. This will make is more reusable and make testing easier